src/callbacks_step/save_restart_dg.jl

```diff
@@ -88,6 +88,66 @@ end
 function save_restart_file(u, time, dt, timestep,
                            mesh::Union{ParallelTreeMesh, ParallelP4estMesh}, equations, dg::DG, cache,
                            restart_callback)
+  if HDF5.has_parallel()
+    save_restart_file_parallel(u, time, dt, timestep, mesh, equations, dg, cache, restart_callback)
+  else
+    save_restart_file_on_root(u, time, dt, timestep, mesh, equations, dg, cache, restart_callback)
+  end
+end
+
+
+function save_restart_file_parallel(u, time, dt, timestep,
+                                    mesh::Union{ParallelTreeMesh, ParallelP4estMesh}, equations, dg::DG, cache,
+                                    restart_callback)
+  @unpack output_directory = restart_callback
+
+  # Filename without extension based on current time step
+  filename = joinpath(output_directory, @sprintf("restart_%06d.h5", timestep))
```

Comment:
```suggestion
  # Filename based on current time step
  filename = joinpath(output_directory, @sprintf("restart_%06d.h5", timestep))
```
It has the extension `.h5`?